SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -74,37 +91,54 @@ SpatialEventTracer::SpatialEventTracer(const FString& WorkerId)
 	const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>();
 	MaxFileSize = Settings->MaxEventTracingFileSizeBytes;
 
-	Trace_EventTracer_Parameters parameters = {};
-	parameters.user_data = this;
-	parameters.callback = &SpatialEventTracer::TraceCallback;
-	EventTracer = Trace_EventTracer_Create(&parameters);
+	Trace_EventTracer_Parameters Parameters = {};
+	Parameters.user_data = this;
+	Parameters.callback = &SpatialEventTracer::TraceCallback;
 
-	Trace_SamplingParameters SamplingParameters = {};
-	SamplingParameters.sampling_mode = Trace_SamplingMode::TRACE_SAMPLING_MODE_PROBABILISTIC;
+	UE_LOG(LogSpatialEventTracer, Log, TEXT("Spatial event tracing enabled."));
 
 	UEventTracingSamplingSettings* SamplingSettings = Settings->GetEventTracingSamplingSettings();
 
-	UE_LOG(LogSpatialEventTracer, Log, TEXT("Setting event tracing sampling probability. Probability: %f."),
-		   SamplingSettings->SamplingProbability);
-
 	TArray<Trace_SpanSamplingProbability> SpanSamplingProbabilities;
-	TArray<std::string> AnsiStrings; // Worker requires ansi const char*
-	for (const auto& Pair : SamplingSettings->EventSamplingModeOverrides)
+	FSpatialTraceEventDataBuilder::FStringCache
+		AnsiStrings; // Storage for strings passed to the worker SDK Worker requires ansi const char*
+
+	if (Settings->bCaptureAllEventTracingData)
 	{
-		const FString& EventName = Pair.Key.ToString();
-		UE_LOG(LogSpatialEventTracer, Log, TEXT("Adding trace event sampling override. Event: %s Probability: %f."), *EventName,
-			   Pair.Value);
-		int32 Index = AnsiStrings.Add(TCHAR_TO_ANSI(*EventName));
-		SpanSamplingProbabilities.Add({ AnsiStrings[Index].c_str(), Pair.Value });
+		UE_LOG(LogSpatialEventTracer, Log, TEXT("Setting event tracing span sampling \"always\"."));
+		Parameters.span_sampling_parameters.sampling_mode = Trace_SamplingMode::TRACE_SAMPLING_MODE_ALWAYS;
```

Comment:
When `bCaptureAllEventTracingData` is `true`, presumably you want all events to be recorded too? Do you not need to set `filter_parameters` here (to "always accept filter") too to ensure this?